docs/rfcs/0561-list-with-metadata.md

```diff
@@ -0,0 +1,72 @@
+- Proposal Name: `list_with_metadata`
+- Start Date: 2022-08-23
+- RFC PR: None
+- Tracking Issue: None
+
+# Summary
+
```

Comment:
Please refer to https://opendal.databend.rs/rfcs/0337-dir-entry.html here so that we know why we have `DirEntry`.